PluginGenerator/Resources/Core/Plugin.java

```diff
@@ -20,6 +20,10 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(Plugin.class);
 
+  protected static final String KEY = "[PLUGIN_KEY]";
+
+  protected static final String NAME = "[PLUGIN_NAME]";
+
```

Comment:
Added some new protected properties to the core plugin Java class so they can be referenced by subclasses.
Previously the subclasses were directly referring to the source code substitution tokens (e.g. _[XXX]_ ) which was more error-prone and harder to debug. Now we'd get compilation errors which are easier to fix.